src/doom/g_game.c

```diff
@@ -2997,6 +2997,10 @@ boolean G_CheckDemoStatus (void)
                 S_Start();
             }
 
+            // [crispy] record demo join
+            if (cmd != NULL)
+                cmd->buttons |= BT_JOIN;
```

Comment:
Extra curly bracket pairs, please.

Reply:
Done

Reply:
What if you reset `last_cmd` back to `NULL` right here instead of doing the switch some lines above?

Reply:
This code won't always be reached when this method is called. So, in that case the `last_cmd` variable wouldn't be set to `NULL`, and then a later call of this method could be using an old version of `last_cmd`. I don't really know enough about the whole picture of the flow to know if that actually makes a difference or is possible, but that's why I set it to `NULL` at the top.

Reply:
We're using this kind of like a manual stack, so I think it's safest to "pop" the value when we enter the function instead of potentially leaving something around. Although it might make no difference now, maybe it would have some side effect in the future that would catch someone off-guard.

Reply:
I see, thank you! Last question, are we sure this works? I mean, is this bit actually written into the joined demo?

It should, because `G_WriteDemoTiccmd()` is called after `G_ReadDemoTiccmd()` in `G_Ticker()` and uses the same `ticcmd_t` as its argument. But is it tested, does it work?

Reply:
Yes, I tested a few cases:
1) Normal demo (nothing set)
2) Joining a demo early by pressing Q during playback (sets bit)
3) Joining a demo at the end of playback (-playback with -record) (sets bit)
4) Joining a demo that already had a join in it (>2 segments) (sets bits at each join spot)

I used my demo analysis tool https://github.com/kraflab/lmp with an extra modification to check for the new data.